src/XrdSys/XrdSysE2T.cc

```diff
@@ -67,11 +67,13 @@ int initErrTable()
    // EAUTH is remapped to EBADE ('invalid exchange').  Given there's no current XRootD use of a
    // syscall that can return EBADE, we assume EBADE really means authentication denied.
 #if defined(EBADE)
-   if (Errno2String[EBADE]) {
-     free((char*)Errno2String[EBADE]);
+   if (EBADE - ERRNOBASE > 0 && EBADE - ERRNOBASE < errSlots) {
+      if (Errno2String[EBADE - ERRNOBASE])
+         free((char*)Errno2String[EBADE - ERRNOBASE]);
+      Errno2String[EBADE - ERRNOBASE] = "authentication failed - possible invalid exchange";
+      if (EBADE - ERRNOBASE > lastGood)
+         lastGood = EBADE - ERRNOBASE;
```

Comment:
As long as e add this, would it not mae sense to add EBADE to E2SMap if it can't be represented in Errno2String[] via an else clause?

Reply:
Implemented.